pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ApexCommentBuilder.java

```diff
@@ -103,7 +104,8 @@ public void buildFormalComment(AbstractApexNode node) {
     }
 
     private static CommentInformation extractInformationFromComments(TextDocument sourceCode, String suppressMarker) {
-        ApexLexer lexer = new ApexLexer(CharStreams.fromString(sourceCode.getText().toString()));
+        String source = sourceCode.getText().toString();
+        ApexLexer lexer = new ApexLexer(new CaseInsensitiveInputStream(CharStreams.fromString(source)));
```

Comment:
We should here also remove the default error listener of antlr and add our own, that really fails instead of just logging to stderr:

```java
        lexer.removeErrorListeners();
        lexer.addErrorListener(new BaseErrorListener() {
            @Override
            public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine, String msg, RecognitionException e) {
                throw new ParseException("line " + line + ":" + charPositionInLine + " " + msg);
            }
        });
```

(Note: ParseException = net.sourceforge.pmd.lang.ast.ParseException)

Otherwise, we won't be able to create an test easily, as we won't react on any lexer error.

Reply:
Something similar is already used elsewhere:
* [AntlrTokenManager](https://github.com/pmd/pmd/blob/main/pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/impl/antlr4/AntlrTokenManager.java#L54), throws `LexException`
* [GroovyTokenManager](https://github.com/pmd/pmd/blob/main/pmd-groovy/src/main/java/net/sourceforge/pmd/lang/groovy/ast/impl/antlr4/GroovyTokenManager.java#L72), throws `LexException`
* [PmdSwiftParser](https://github.com/pmd/pmd/blob/main/pmd-swift/src/main/java/net/sourceforge/pmd/lang/swift/ast/PmdSwiftParser.java#L29), only logs, TODO for throwing `ParseException`

Perhaps that custom error listener can be refactored into a dedicated class and reused in all cases.

Reply:
That's a good observation. However, I'd prefer to keep this PR limited and try to avoid growing it further.

These refactorings you mention makes sense, but should be done in a separate step - separate PR. Note, that we deliberately don't throw an exception in PmdSwiftParser, as the grammar is far from complete - throwing an exception here basically means, that only few Swift code can be parsed.
Note, that we should keep throwing a ParseException instead of a LexException in ApexCommentBuilder, as this is called from the PMD "check" goal rather than from CPD (copy paste detection).
Looking at the other examples, the exception should be probably build with:

```java
throw new ParseException(msg).withLocation(FileLocation.caret(sourceCode.getFileId(), line, charPositionInLine));
```